### PR TITLE
Spdx document file handling improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt [hunk 1]
```diff
@@ -350,7 +350,7 @@ class SpdxDocumentFile(
         val externalDocumentReference = doc.externalDocumentRefs.find {
             it.externalDocumentId == documentRef
         } ?: throw IllegalArgumentException(
-            "No single package or externalDocumentRef with ID '$identifier' found."
+            "ID '$identifier' could neither be resolved as a package nor as an externalDocumentRef."
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt:350]: "resolved to" instead of "resolved as".

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt:350]: I was using "as" intentionally, but I'm also fine with "to". Will change in both places.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt:350]: Or did you really only want to get "as" changed to "to" in the first occurrence?

Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt:350]: I meant both occurrences of "as", I'm not a native speaker but "resolve as" just sounds wrong to me.

Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt:350]: Anyway, if you want to keep it as is that's fine with me.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt]: Changed in both places.

**Review by fviernau — APPROVED** [review]

**Pull request merged**